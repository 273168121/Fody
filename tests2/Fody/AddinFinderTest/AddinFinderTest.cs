using System.IO;
using System.Linq;
using Xunit;
#if NET46 // TODO: Remove when ObjectApproval supports .NET Core
using ObjectApproval;
#endif

public partial class AddinFinderTest : TestBase
{
    [Fact]
    public void WithNuGetPackageRoot()
    {
        var combine = Path.GetFullPath(Path.Combine(AssemblyLocation.CurrentDirectory, "Fody/FakeNuGetPackageRoot"));
        var nuGetPackageRoot = Path.GetFullPath(combine);
        var result = AddinFinder.ScanDirectoryForPackages(nuGetPackageRoot)
            .Select(s => s.Replace(@"\\", @"\").Replace(combine, ""))
            .ToList();
#if NET46 // TODO: Remove when ObjectApproval supports .NET Core
        ObjectApprover.VerifyWithJson(result);
#endif
    }

    [Fact]
    public void Integration_OldNugetStructure()
    {
        var combine = Path.GetFullPath(Path.Combine(AssemblyLocation.CurrentDirectory, "Fody/AddinFinderTest/OldNugetStructure"));
        Verify(combine);
    }

    [Fact]
    public void Integration_NewNugetStructure()
    {
        var combine = Path.GetFullPath(Path.Combine(AssemblyLocation.CurrentDirectory, "Fody/AddinFinderTest/NewNugetStructure"));
        Verify(combine);
    }

    [Fact]
    public void Integration_PaketStructure()
    {
        var combine = Path.GetFullPath(Path.Combine(AssemblyLocation.CurrentDirectory, "Fody/AddinFinderTest/PaketStructure"));
        Verify(combine);
    }

    static void Verify(string combine)
    {
        var addinFinder = new AddinFinder(
            log: s => { },
            solutionDirectory: Path.Combine(combine, "Solution"),
            msBuildTaskDirectory: Path.Combine(combine, "MsBuildDirectory/1/2/3"),
            nuGetPackageRoot: Path.Combine(combine, "NuGetPackageRoot"),
            packageDefinitions: null);
        addinFinder.FindAddinDirectories();
#if NET46 // TODO: Remove when ObjectApproval supports .NET Core
        ObjectApprover.VerifyWithJson(addinFinder.FodyFiles.Select(x => x.Replace(combine, "")));
#endif
    }
}