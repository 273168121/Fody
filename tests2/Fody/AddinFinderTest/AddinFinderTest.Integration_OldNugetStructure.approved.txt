[
  "\\Solution\\Packages\\Weaver.Fody.7.0.0\\netclassicweaver\\Weaver.Fody.dll",
  "\\Solution\\Packages\\Weaver.Fody.8.0.0\\Weaver.Fody.dll",
  "\\FromNugetConfig\\Weaver.Fody.1.0.0\\Weaver.Fody.dll",
  "\\FromNugetConfig\\Weaver.Fody.2.0.0\\netclassicweaver\\Weaver.Fody.dll",
  "\\MsBuildDirectory\\Weaver.Fody.3.0.0\\Weaver.Fody.dll",
  "\\MsBuildDirectory\\Weaver.Fody.4.0.0\\netclassicweaver\\Weaver.Fody.dll",
  "\\Solution\\Tools\\Weaver.Fody.dll",
  "\\Solution\\Tools\\Weaver.Fody.10.0.0\\netclassicweaver\\Weaver.Fody.dll",
  "\\Solution\\Tools\\Weaver.Fody.9.0.0\\Weaver.Fody.dll",
  "\\NuGetPackageRoot\\Weaver.Fody.5.0.0\\Weaver.Fody.dll",
  "\\NuGetPackageRoot\\Weaver.Fody.6.0.0\\netclassicweaver\\Weaver.Fody.dll"
]